data-pipeline/telemetry-location-updater/src/main/java/org/ekstep/ep/samza/util/UserLocationCache.java

```diff
@@ -0,0 +1,112 @@
+package org.ekstep.ep.samza.util;
+
+import com.datastax.driver.core.Row;
+import org.apache.samza.config.Config;
+import org.ekstep.ep.samza.core.Logger;
+import org.ekstep.ep.samza.domain.Location;
+import com.datastax.driver.core.querybuilder.*;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.exceptions.JedisException;
+
+public class UserLocationCache {
+
+    private static Logger LOGGER = new Logger(UserLocationCache.class);
+
+    private String cassandra_db;
+    private String cassandra_user_table;
+    private String cassandra_location_table;
+    private Config config;
+    private RedisConnect redisConnect;
+    private CassandraConnect cassandraConnection;
+    private int locationDbKeyExpiryTimeInSeconds;
+
+    public UserLocationCache(Config config, RedisConnect redisConnect, CassandraConnect cassandraConnect) {
+        this.redisConnect = redisConnect;
+        this.config = config;
+        this.cassandra_db = config.get("middleware.cassandra.keyspace", "sunbird");
+        this.cassandra_user_table = config.get("middleware.cassandra.user_table", "user");
+        this.cassandra_location_table = config.get("middleware.cassandra.location_table", "location");
+        this.cassandraConnection = cassandraConnect;
+        this.locationDbKeyExpiryTimeInSeconds = config.getInt("location.db.redis.key.expiry.seconds", 86400);
+    }
+
+    public Location getLocationByUser(String userId) {
+        if (userId == null) return null;
+
+        try (Jedis jedis = redisConnect.getConnection()) {
```

Comment:
Use a different database in redis to store the user location cache.

Reply:
Now for user cache redis uses database no. 1